cpp/oneapi/dal/backend/primitives/stat/cov_dpc.cpp

```diff
@@ -79,13 +80,41 @@ inline sycl::event compute_covariance(sycl::queue& q,
         cgh.depends_on(deps);
         cgh.parallel_for(range, [=](sycl::item<2> id) {
             const std::int64_t gi = id.get_linear_id();
-            const std::int64_t i = id.get_id(0);
-            const std::int64_t j = id.get_id(1);
 
-            if (i < p && j < p) {
-                cov_ptr[gi] -= inv_n * sums_ptr[i] * sums_ptr[j];
-                cov_ptr[gi] *= multiplier;
-            }
+            cov_ptr[gi] -= inv_n * sums_ptr[id.get_id(0)] * sums_ptr[id.get_id(1)];
+            cov_ptr[gi] *= multiplier;
+        });
+    });
+}
+
+template <typename Float>
+inline sycl::event compute_covariance_centered(sycl::queue& q,
```

Comment:
Please add the description. It is not obvious that on input `cov` is a matrix of sum of squares and on output it is the covariance matrix.
The same applies to `compute_covariance`.